### PR TITLE
[INTEL oneDNN] Refactoring: rename and enhance IsMKLEnabled Method

### PR DESCRIPTION
**Pull request by gzmkl**
1. Rename IsMKLEnabled() (C++ API) to IsOneDNNEnabled() as "MKL DNN" has been replaced by "OneDNN".

2. Enhancement IsOneDNNEnabled() implementation: Remove the use of environment variable TF_DISABLE_MKL as it has been replaced by TF_ENABLE_ONEDNN_OPTS.

### CODE REVIEW

**Comment by gzmkl** [comment]
@qqfish Hi, merge conflicts should have been addressed. Please help review. Thanks!

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ Status CreateUnshapedOutput(
 #if defined(IS_MOBILE_PLATFORM)
   return errors::Unimplemented(
       "Remote outputs are not available on mobile devices.");
-#else  // !IS_MOBILE_PLATFORM
+#else   // !IS_MOBILE_PLATFORM
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:1259]: Nit: Please remove this extra space. (The original code has two spaces between `#else` and `//`, which is already correct.) Some for other places.

(Did you use the wrong formatter option for just this file? Other file seems to be fine.)

File: tensorflow/core/common_runtime/mkl_tfconversion_pass.cc [hunk 2]
```diff
@@ -425,8 +425,8 @@ Status MklToTfConversionPass::Run(const GraphOptimizationPassOptions& options) {
   if (options.graph == nullptr && options.partition_graphs == nullptr) {
     return Status::OK();
   }
-  if (!IsMKLEnabled()) {
-    VLOG(2) << "TF-MKL: MKL is not enabled";
+  if (!IsOneDNNEnabled()) {
+    VLOG(2) << "TF-MKL: oneDNN is not enabled";
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_tfconversion_pass.cc:429]: Nit:
```suggestion
    VLOG(2) << "TF-oneDNN: oneDNN is not enabled";
```

File: tensorflow/core/grappler/optimizers/remapper.cc [hunk 3]
```diff
@@ -286,8 +286,8 @@ bool HasDataType(const NodeDef* node, const DataType& expected,
 bool IsCpuCompatibleDataType(const NodeDef* contraction,
                              const string& type_attr = "T") {
   DataType dtype = GetDataTypeFromAttr(*contraction, type_attr);
-  // Stock TF without oneDNN build will always be `false`.
-  bool is_one_dnn_enabled = IsMKLEnabled();
+  // Stock TensorFlow is built with oneDNN enabled as default.
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/remapper.cc:289]: Nit:
```suggestion
  // Stock TensorFlow is built with oneDNN, enabled by default on some CPUs.
```

File: tensorflow/core/util/util.cc [hunk 4]
```diff
@@ -127,24 +127,23 @@ string SliceDebugString(const TensorShape& shape, const int64_t flat) {
   return result;
 }
 
-bool IsMKLEnabled() {
+bool IsOneDNNEnabled() {
+  // For platforms which oneDNN does support, always return false.
```
Comment (penpornk) [inline tensorflow/core/util/util.cc:131]: Nit:
```suggestion
  // For platforms which oneDNN doesn't support, always return false.
```

File: tensorflow/core/util/util.cc [hunk 5]
```diff
@@ -127,24 +127,23 @@ string SliceDebugString(const TensorShape& shape, const int64_t flat) {
...
+    return true;
+#endif  // ENABLE_MKL
+
+  // Stock TensorFlow
   // Linux: Turn oneDNN on by default for CPUs with neural network features.
   // Windows: oneDNN is off by default.
-  // No need to guard for other platforms here because INTEL_MKL is only defined
-  // for non-mobile Linux or Windows.
+  // No need to guard for other platforms because INTEL_MKL is only defined
+   // for non-mobile Linux or Windows.
```
Comment (penpornk) [inline tensorflow/core/util/util.cc:145]: Nit: Please remove the extra space.
```suggestion
  // for non-mobile Linux or Windows.
```

File: tensorflow/core/util/util.cc [hunk 6]
```diff
@@ -127,24 +127,23 @@ string SliceDebugString(const TensorShape& shape, const int64_t flat) {
...
+  static absl::once_flag once;
```
Comment (penpornk) [inline tensorflow/core/util/util.cc:146]: This flag is not used. Please remove?

**Review by penpornk — CHANGES_REQUESTED** [review]
Sorry for the delay!

File: tensorflow/core/util/util.cc (hunk 6 above)
Comment (gzmkl) [reply tensorflow/core/util/util.cc:146]: The flag is used in line 158.  I am going to move the line to just where it is used.

Comment (gzmkl) [reply tensorflow/core/util/util.cc]: Done

File: tensorflow/core/util/util.cc (hunk 5 above)
Comment (gzmkl) [reply tensorflow/core/util/util.cc]: code change done

File: tensorflow/core/util/util.cc (hunk 4 above)
Comment (gzmkl) [reply tensorflow/core/util/util.cc]: code change done

File: tensorflow/core/common_runtime/mkl_tfconversion_pass.cc (hunk 2 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_tfconversion_pass.cc]: code change done

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 1 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/eager/execute.cc]: I do not recall this change, maybe an accident. Just change it back

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 7]
```diff
@@ -1294,12 +1294,12 @@ Status AddOrExecuteNode(core::RefCountPtr<KernelAndDevice> kernel,
...
   absl::optional<EagerFunctionParams> eager_func_params =
       op->eager_func_params();
+  // Create an eager op id for a cross-process function if not exist.
   if (kernel->IsCrossProcess() && !eager_func_params.has_value()) {
-    // Create an eager op id for a cross-process function if not exist.
 #if defined(IS_MOBILE_PLATFORM)
     return errors::Unimplemented(
         "Cross-process functions are not supported on mobile devices.");
-#else  // !IS_MOBILE_PLATFORM
+#else   // !IS_MOBILE_PLATFORM
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:1302]: Nit: There are two more places
```suggestion
#else  // !IS_MOBILE_PLATFORM
```

File: tensorflow/core/common_runtime/eager/execute.cc [hunk 8]
```diff
@@ -1721,7 +1721,7 @@ Status GetKernelOutputs(
 #if defined(IS_MOBILE_PLATFORM)
         return errors::Unimplemented(
             "Remote outputs are not available on mobile devices.");
-#else  // !IS_MOBILE_PLATFORM
+#else   // !IS_MOBILE_PLATFORM
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/execute.cc:1724]: Nit:
```suggestion
#else  // !IS_MOBILE_PLATFORM
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes!

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 8 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/eager/execute.cc]: done

File: tensorflow/core/common_runtime/eager/execute.cc (hunk 7 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/eager/execute.cc]: done

**Review by penpornk — APPROVED** [review]
Thank you again!

**Comment by gzmkl** [comment]
@penpornk I noticed quite a few check failures and did not see any specific error caused by this PR change. Please help re-run the tests, or point out the specific errors which I need to take care of. Thanks!

**Comment by penpornk** [comment]
@gzmkl Sorry for the delay! Could you please help resolve the conflicts?

**Comment by ashiqimranintel** [comment]
@penpornk , any comments on this PR?

**Comment by ashiqimranintel** [comment]
@penpornk , can you re-run the test?

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 9]
```diff
@@ -4194,10 +4194,10 @@ Status MklLayoutRewritePass::Run(const GraphOptimizationPassOptions& options) {
...
     return OkStatus();
   }
-  if (!IsMKLEnabled()) {
-    VLOG(2) << "TF-MKL: MKL is not enabled";
-    return OkStatus();
-  }
+
+  if (!IsOneDNNEnabled()) {
+    VLOG(2) << "TF-MKL: oneDNN is not enabled";
+    return Status::OK();
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:4200]: I think we are missing a closing bracket (`}`).

File: tensorflow/core/grappler/optimizers/remapper.cc [hunk 10]
```diff
@@ -1354,7 +1357,8 @@ bool FindMatMulBiasAddAndGelu(RemapperContext* ctx, int node_index,
 
     // matmul_node is already the _FusedMatMul and we don't need to check its
     // data type again.
-    if (!IsMKLEnabled() && !NodeIsOnGpu(matmul_node)) return false;
+
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/remapper.cc:1360]: Please remove the extra blank line.

**Review by penpornk — CHANGES_REQUESTED** [review]
Sorry again for the delay!

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 9 above)
Comment (gzmkl) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc:4200]: Must be introduced when addressing merge conflicts (from other oneDNN PRs). Fixed.

Thanks!

File: tensorflow/core/grappler/optimizers/remapper.cc (hunk 10 above)
Comment (gzmkl) [reply tensorflow/core/grappler/optimizers/remapper.cc]: Done

**Comment by gzmkl** [comment]
Just resolved another merge conflict...

**Comment by gbaned** [comment]
Hi @gzmkl Can you please review this PR ? Thank you!

**Comment by gzmkl** [comment]
Addressed merge conflict related to Penporn's recent PR

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Review by ezhulenev — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Review by ezhulenev — APPROVED** [review]

**Comment by gzmkl** [comment]
@penpornk @ezhulenev  Please hold on reviewing this PR. Due to a lot of merge conflicts, I am working on a new PR which is based on the latest master (with oneDNN 3.x fully integrated, all conflicts will be gone).   

Thanks

**Comment by gbaned** [comment]
Hi @gzmkl Any update on this CL? Please. Thank you!

**Comment by gbaned** [comment]
Hi @gzmkl Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @gzmkl Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @gzmkl Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @gzmkl I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen. Thank you for your contribution!

**Pull request closed without merging**